melodies_monet/plots/aircraftplots.py

```diff
@@ -0,0 +1,842 @@
+
+                # Box properties
+                boxprops = {
+                    'facecolor': 'none',  # Transparent face color
+                    'edgecolor': color,   # Edge color matching the median line color
+                    'linewidth': 2        # Thickness of the box
+                }
+
+                # Whisker properties
+                whiskerprops = {
+                'color': color,       # Whisker color
+                'linewidth': 2        # Thickness of the whiskers
+                }
+
+                # Median line properties
+                medianprops = {
+                    'color': 'none'       # Transparent color for median line
+                }
+
+                capprops = {'color': color, 'linewidth': 2} # Whisker end bar properties
+
+                # Create the box plot
+                ax.bxp([data], vert=False, positions=[binmidpoint.values[idx]],
+                       widths=500, meanline=True, patch_artist=True, 
+                       boxprops=boxprops, whiskerprops=whiskerprops, medianprops=medianprops,
+                      capprops=capprops)
+
+                # Optional: Add a legend entry for the box (only once)
+                if legend_label and idx == 0:
+                    box_legend_artist = Rectangle((0, 0), 1, 1, facecolor='none', edgecolor=color, linewidth=2)
+                    ax.legend([box_legend_artist], [f"{legend_label} (box-whisker)"], loc='upper left')
+            
+
+        plot_kwargs['label'] = f"{label} (median)"
+        median_line_df = pd.DataFrame(data={'median': median.values, 'binmidpoint': binmidpoint.values})
+        ax = median_line_df.plot(x='median', y='binmidpoint', ax=ax, legend=True, **plot_kwargs)
+        #ax= median_line_df.plot(ax=ax, legend=True, **plot_kwargs)
+        
+                
+    
+    # If plot has been created, add to the current axes
+    else:
+        # This means that an axis handle already exists, so use it to plot the model output
+        altitude_bins = pd.cut(df[altitude_variable], bins=bins)
+        # Calculate the midpoints of the altitude bins
+        bin_midpoints = altitude_bins.apply(lambda x: x.mid)
+        # Convert bin_midpoints to a column in the DataFrame
+        df['bin_midpoints'] = bin_midpoints
+        # can be .groupby(bin_midpoints) as well (qzr)
+        median = df.groupby(altitude_bins)[column].median()
+        q1 = df.groupby(altitude_bins)[column].quantile(0.25)
+        q3 = df.groupby(altitude_bins)[column].quantile(0.75)
+        # Convert bin_midpoints to a numerical data type
+        df['bin_midpoints'] = df['bin_midpoints'].astype(float)
+
+        # Calculate the 10th, 90th, 5th, and 95th percentiles
+        p10 = df.groupby(altitude_bins)[column].quantile(0.10)
+        p90 = df.groupby(altitude_bins)[column].quantile(0.90)
+        p5 = df.groupby(altitude_bins)[column].quantile(0.05)
+        p95 = df.groupby(altitude_bins)[column].quantile(0.95)
```

Comment:
`p5` and `p95` are never used.
```suggestion
        p10 = df.groupby(altitude_bins)[column].quantile(0.10)
        p90 = df.groupby(altitude_bins)[column].quantile(0.90)
```